src/CoreWCF.Http/src/CoreWCF/Channels/HttpTransportBindingElement.cs

```diff
@@ -222,6 +224,27 @@ public override T GetProperty<T>(BindingContext context)
             {
                 return (T)(object)new HttpTransportServiceBuilder();
             }
+
+            if (typeof(T) == typeof(IAuthorizationCapabilities))
+            {
+                var binding = context.BindingParameters.OfType<Binding>().FirstOrDefault();
+                if (binding is HttpBindingBase httpBindingBase)
+                {
+                    context.BindingParameters.Remove(httpBindingBase);
+                    return (T)(object)new DefaultAuthorizationCapabilities(httpBindingBase?.BasicHttpSecurity.Transport.ClientCredentialType ==
+                                                                           HttpClientCredentialType.InheritedFromHost);
+                }
+                if (binding?.GetType().FullName == "CoreWCF.WebHttpBinding")
+                {
+                    context.BindingParameters.Remove(binding);
+                    dynamic webHttpBinding = binding;
```

Comment:
>dynamic webHttpBinding = binding; [](http://example.com/codeflow?start=20&length=33)

 It's been a while since I deep a deep dive on how the dynamic keyword works and I forgot you could do things like that. If you look at the generated IL it's an expensive operation with a lot of stuff behind that one keyword so it should be used sparingly. We might need to remove it in the future too and find another way to do this (add a property on HttpTransportBindingElement?) as asp.net core is working to add support for aot compilation and we might want to follow. I have a suspicion this will break aot compilation. It's okay for now, just saying we need to minimize usage overall.

Reply:
Totally agree. I will create a separate issue to remediate this.

Reply:
finally done in this [commit](https://github.com/CoreWCF/CoreWCF/pull/850/commits/3c2d4b92aee4bc625b4275da639172a0678d661d)